modules/etcd-creating-automated-backups.adoc

```diff
@@ -0,0 +1,320 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.adoc
+
+:_content-type: PROCEDURE
+[id="creating-automated-etcd-backups_{context}"]
+= Creating automated etcd backups
+
+The automated backup feature for etcd supports both recurring and single backups. Recurring backups create a cronjob that starts a single backup each time it triggers.
+
+You can set the method of backup retention to `RetentionNumber`, `RetentionSize`, or `""`. If unspecified or set to `""`, the default retention type is `RetentionSize` with 15 maximum backups. Specifying a retention type overrides the default value.
+
+Use dynamically-provisioned storage to keep the created etcd backup data in a safe, external location if possible. If dynamically-provisioned storage is not available, consider storing the backup data on an NFS share to make backup recovery more accessible.
+
+:FeatureName: Automating etcd backups
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have access to the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Create a `FeatureGate` custom resource (CR) file named `etcd-enable-automated-backups.yaml` with the following contents:
++
+[WARNING]
+====
+The `CustomNoUpgrade` feature set required for this procedure is **not supported**, **cannot be undone**, and **prevents upgrades**.
+====
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: FeatureGate
+metadata:
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    release.openshift.io/create-only: "true"
+  name: cluster
+spec:
+  customNoUpgrade:
+    enabled:
+      - AutomatedEtcdBackup
+  featureSet: CustomNoUpgrade
+----
+
+. Run the following command to apply the CR and enable automated backups:
++
+[source,terminal]
+----
+$ oc apply -f etcd-enable-automated-backups.yaml
```

Comment:
it takes a little time to enable the APIs after running that command. So immediately posting the backup resource will certainly fail. Could you leave a quick note on that as well, please?

Reply:
Is there a command that the user can run to verify whether or not the APIs are enabled, or do we just have to tell them to wait a bit?

Reply:
seems this is documented at the bottom of https://docs.openshift.com/container-platform/4.13/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-cli_nodes-cluster-enabling

not sure that's the right indication for us though. I think we can also check whether the api CRD exists. 
Something along the lines of `oc get crd | grep etcdbackup` - I don't have a cluster handy to test right now, will let you know once it's working.

Reply:
so the error you'll get back when it's not installed yet is:
> error: resource mapping not found for name: "testbackup" namespace: "openshift-etcd" from "/home/tjungblu/Downloads/backup.yaml": no matches for kind "EtcdBackup" in version "operator.openshift.io/v1alpha1"
ensure CRDs are installed first

(the file, names and namespace obviously differs)

> $ oc get crd | grep backup

indeed works to test whether the CRD is installed.

It didn't work after posting the feature gate as `CustomNoUpgrade` though, seems the API installation is gated on `TechPreviewNoUpgrade`. So the patch would look like what's described in:

https://docs.openshift.com/container-platform/4.13/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-cli_nodes-cluster-enabling

```yaml
---
apiVersion: config.openshift.io/v1
kind: FeatureGate
metadata:
  name: cluster
spec:
  featureSet: TechPreviewNoUpgrade
```

Enabling the flag with `CustomNoUpgrade` seems to enable the feature in cluster-etcd-operator, but does not install the CRDs. Still takes a little while, but it eventually shows up.